src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final String SPLITTER = " ";
+    private static final String HYPHEN = " - ";
+    private final Map<String, Integer> employees = new HashMap<>();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        for (String name : names) {
+            employees.put(name, 0);
+        }
+        StringBuilder reportForPeriod = new StringBuilder("Report for period ").append(dateFrom)
+                .append(HYPHEN).append(dateTo);
+        for (String employeeData : data) {
```

Comment:
your solution is good, the only problem is that checklist says that we should not use Map
try solving this problem with 2 loops:

```
// attach header to StringBuilder
for (String name : names) {
    for (String employeeData : data) {
       // calculate salary for each row, add result to some `total` variable
    }
    // add total for each name to builder
}